### PR TITLE
Get XContent params from request in Nodes rest actions

### LINKED ISSUE #18794: flat_settings ignored on nodes info in 5.0.0-alpha3
**Elasticsearch version**: 5.0.0-alpha3

**OS version**: Windows 10

**Description of the problem including expected versus actual behavior**:

`flat_settings` is not respected on nodes info.

**Steps to reproduce**:
 1. call `/_nodes/_all/settings?flat_settings=true&timeout=2s&pretty=true`
 2. observe the settings response is not flat e.g.

```json
{
  "_nodes" : {
    "total" : 1,
    "successful" : 1,
    "failed" : 0
  },
  "cluster_name" : "sniffroledetection-cluster-b857db",
  "nodes" : {
    "0f3jW5LyR4K2lYQl8arDAw" : {
      "name" : "sniffroledetection-node-b857db",
      "transport_address" : "127.0.0.1:9300",
      "host" : "127.0.0.1",
      "ip" : "127.0.0.1",
      "version" : "5.0.0-alpha3",
      "build_hash" : "cad959b",
      "http_address" : "127.0.0.1:9200",
      "roles" : [ "master", "ingest" ],
      "attributes" : {
        "testingcluster" : "true"
      },
      "settings" : {
        "cluster" : {
          "name" : "sniffroledetection-cluster-b857db"
        },
        "node" : {
          "name" : "sniffroledetection-node-b857db",
          "attr" : {
            "testingcluster" : "true"
          },
          "data" : "false",
          "master" : "true"
        },
        "path" : {
          "logs" : "C:/Users/russ/AppData/Roaming/NEST/5.0.0-alpha3/elasticsearch-5.0.0-alpha3/logs",
          "home" : "C:\\Users\\russ\\AppData\\Roaming\\NEST\\5.0.0-alpha3\\elasticsearch-5.0.0-alpha3",
          "repo" : "C:\\Users\\russ\\AppData\\Roaming\\NEST\\5.0.0-alpha3\\repositories"
        },
        "client" : {
          "type" : "node"
        },
        "threadpool" : {
          "watcher" : {
            "queue_size" : "1000",
            "size" : "40",
            "type" : "fixed"
          }
        },
        "script" : {
          "inline" : "true",
          "stored" : "true"
        },
        "xpack" : {
          "security" : {
            "enabled" : "false"
          }
        }
      }
    }
  }
}
```

`flat_settings` is honored in 5.0.0-alpha1 and 5.0.0-alpha2 so it appears to be a regression in 5.0.0-alpha3.

### PR DESCRIPTION
**Pull request by alexshadow007**
Closes #18794

### CODE REVIEW

**Comment by nik9000** [comment]
Fix looks good. Can you add a test for it? Something in the rest tests (`rest-api-spec/src/main/resources/rest-api-spec/test/`probably.

You can run with with `gradle distribution:integ-test-zip:check`.

**Comment by alexshadow007** [comment]
@nik9000 done

File: rest-api-spec/src/main/resources/rest-api-spec/test/nodes.info/30_settings.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        metric: [ settings ]
+
+  - match : { nodes.$master.settings.client.type: node }
+
+  - do:
+      nodes.info:
+        metric: [ settings ]
+        flat_settings: true
+
+  - match : { nodes.$master.settings.client\.type: node }
```
Comment (nik9000) [inline rest-api-spec/src/main/resources/rest-api-spec/test/nodes.info/30_settings.yaml]: Nice

**Comment by nik9000** [comment]
Looks good. I'll test and merge!

**Comment by nik9000** [comment]
> Looks good. I'll test and merge!

And the tests passed. Thanks so much! Merged!

**Pull request merged**